File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        int res = 0;
        for(int i=0;i<s.length(); i++){
            int[] alpha = new int[26];
            for(int j=i;j < s.length(); j++){
                alpha[s.charAt(j) - 'a']++;
                res += beauty(alpha);
            }
        }
        return res;
    }
    public int beauty(int[] alpha){
        int mf = -1;
        int lf = Integer.MAX_VALUE;
        for(int i=0;i<alpha.length; i++){
            mf = Math.max(mf, alpha[i]);
            if(alpha[i]>=1){
                lf = Math.min(lf, alpha[i]);
            }
        }
        return mf-lf;
    }
}

File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.py
class Solution:
    def beautySum(self, s: str) -> int:
        n = len(s)
        res = 0
        for i in range(n):
            freq = defaultdict(int)
            for j in range(i, n):
                freq[s[j]] += 1
                vals = freq.values()
                res += max(vals) - min(vals)
        return res